icu4j/main/classes/core/src/com/ibm/icu/impl/units/UnitConverter.java

```diff
@@ -128,7 +128,7 @@ public static Factor processFactor(String factor) {
             assert (!factor.isEmpty());
 
             // Remove all spaces in the factor
-            factor.replaceAll("\\s+", "");
```

Comment:
This is obviously a coding error. `replaceAll()` is not changing `String factor` in place (String is immutable anyway).